### DISCUSSION TITLE
好书推荐

### DISCUSSION THREAD

**Discussion by yanglbme** [doocs/technical-books#30] [Ideas]
> 读书百遍，其义自见。

很欢迎各位同学来推荐、分享好的技术类书籍，大家一起阅读。可以在 issues 区推荐，也可以直接提交 PR。

**Comment by purechamploo** [comment]
码农翻身
编码-隐藏在计算机背后的语言

**Comment by yanglbme** [comment]
@purechamploo 谢谢推荐，待会儿会更新书单，添加这两本书籍。

**Comment by yanglbme** [comment]
为了避免项目文件过大，下载过慢，本项目所有书籍资源均采用网盘形式共享。

**Comment by kginglam** [comment]
可以放一点DDD、UML之类的书籍上去，我觉得这些思想类书籍也是挺不错的，敲代码的能看、做产品的也能看

**Comment by yanglbme** [comment]
不错的建议，谢谢反馈 @kginglam

**Comment by rockyyu66** [comment]
两本国内大牛写的Java好书：   
《实战Java高并发程序设计（第2版）》  葛一鸣    
《Java设计模式》刘伟

**Comment by yanglbme** [comment]
@cellei 葛一鸣这本书有看过，确实写得不错，待会儿更新书籍清单。

**Comment by LGSKOKO** [comment]
《Java编程的逻辑》这本书也挺好的，评价也挺高的，个人感觉比《Java核心技术》更贴切些；  作者：马俊昌

**Comment by Wnlife-Li** [comment]
这个仓库很棒

**Comment by caopengan** [comment]
网盘地址怎么都没了？

**Comment by cjq46737** [comment]
Java Concurrency in Practice
现在读的翻译版，相信原版更优秀

**Comment by xdmdcp** [comment]
**[《网络是怎样连接的》](https://www.ituring.com.cn/book/1758)**  户根勤 (作者)  周自恒 (译者)

> 本书以探索之旅的形式，从在**浏览器中输入网址**开始，一路追踪了到显示出网页内容为止的整个过程，以图配文，讲解了网络的全貌，并**重点介绍了实际的网络设备和软件是如何工作**的。

望采纳。

**Comment by muyids** [comment]
where is the network pan address?

**Comment by yanglbme** [comment]
@caopengan @muyids 

因部分同学觉得可能侵犯版权，所以目前移除了主页的资源链接。

Git 历史记录仍有保留，https://github.com/doocs/technical-books/blob/b3fd88c6b39efad1b2235a1f40d26d7be7a0646d/README.md

**Comment by ChunelFeng** [comment]
个人觉得很赞的两本书是：【深入详解stl】（看的时候比较赞，但是现在有点老了，也没续集）和【effective C++】

**Comment by patrickhu1** [comment]
《重构 改善既有代码的设计》《代码整洁之道》
代码技巧类，如何保持一个项目的可维护性

**Comment by josexy** [comment]
是否应该加入《编译原理》之类的书籍，比如虎书和龙书。虽然阅读起来较难，但是里面很多知识是挺重要的

**Comment by yanglbme** [comment]
>  是否应该加入《编译原理》之类的书籍，比如虎书和龙书。虽然阅读起来较难，但是里面很多知识是挺重要的

可以的

**Comment by YangFong** [comment]
看着前端类型都是 JS 书籍，我来推荐一本 CSS 吧。

前端 - [《CSS 揭秘》](https://book.douban.com/subject/26745943/)

**Comment by mvpsheng** [comment]
建议在一段文字开头空出两个字符

**Reply by yanglbme** [reply]
hah, markdown 文档一般不做首行缩进

**Reply by mvpsheng** [reply]
好的主要是最近论文才结束，对这个缩进比较敏感，哈哈，我觉得可以加入华章的《软件工程》这本书怎么样？

**Comment by ilovehu** [comment]
Kali linux 高级渗透测试
